### ISSUE TITLE
Dependencies of conda-forget repodata and index.json are out of sync

### ISSUE THREAD

**Issue by ZzEeKkAa** [conda-forge/libhwloc-feedstock#76]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

Dependencies version restriction are different for `libxml2` in conda-forge channel and package metadata:
- `libxml2 >=2.11.5,<3.0.0a0` in conda-forge channel
- `libxml2 >=2.11.5,<2.12.0a0` in package metadata

```shell
conda search libhwloc=2.9.3=default_h554bfaf_1009 --info
```
```
Loading channels: done
libhwloc 2.9.3 default_h554bfaf_1009
------------------------------------
file name   : libhwloc-2.9.3-default_h554bfaf_1009.conda
name        : libhwloc
version     : 2.9.3
build       : default_h554bfaf_1009
build number: 1009
size        : 2.5 MB
license     : BSD-3-Clause
subdir      : linux-64
url         : https://conda.anaconda.org/conda-forge/linux-64/libhwloc-2.9.3-default_h554bfaf_1009.conda
md5         : f36ddc11ca46958197a45effdd286e45
timestamp   : 2023-09-12 15:30:03 UTC
dependencies:
  - libgcc-ng >=12
  - libstdcxx-ng >=12
  - libxml2 >=2.11.5,<3.0.0a0
```

```shell
cat info/index.json
```
```json
{
  "arch": "x86_64",
  "build": "default_h554bfaf_1009",
  "build_number": 1009,
  "depends": [
    "libgcc-ng >=12",
    "libstdcxx-ng >=12",
    "libxml2 >=2.11.5,<2.12.0a0"
  ],
  "license": "BSD-3-Clause",
  "name": "libhwloc",
  "platform": "linux",
  "subdir": "linux-64",
  "timestamp": 1694532603730,
  "version": "2.9.3"
}
```

### Installed packages

```shell
n/a
```

### Environment info

```shell
n/a
```

**Comment by ZzEeKkAa** [comment]
Thank you @jan-janssen for the quick response. It may be related, but more important question, how is it possible to have out of sync in metadata? Package was uploaded 4 month ago last time

**Comment by jan-janssen** [comment]
I feel this is a misunderstanding. The `libhwloc` package was assumed to be compatible to `libxml2 >=2.11.5,<3.0.0a0` this is why we added this in the `meta.yaml` file. In conda-forge `libxml2` was fixed to `2.11`. As this central fix was updated to `2.12` this broke the tests https://github.com/conda-forge/libhwloc-feedstock/pull/71 . To my understanding this is fixed in `libhwloc=2.10` that is why I would like to merge https://github.com/conda-forge/libhwloc-feedstock/pull/72 . 

In summary the statement `libxml2 >=2.11.5,<3.0.0a0` is wrong, as it was not possible to update to `2.12`.
